Handle youtube errors better

binaryoverload opened the pull request:
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] Tested changes

## What is added
3 tries to connect to YT
New exception to distinguish between a failure and a non-existent video/playlist

## What is fixed
Update apache commons lang so everything works
Update JDA

WalshyDev commented on src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java:
```diff
@@ -72,4 +82,25 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
+        for (int i = 0; i <= 2; i++) {
+            try {
+                item = Optional.ofNullable(player.resolve(input));
+                failed = false;
```
Not even a point setting failed here as it breaks. This just wastes a bit of operation time, macroseconds yes but still operation time.

binaryoverload replied:
It needs that to distinguish between a failure and a non-existent video

ArsenArsen replied:
It is already false by default

binaryoverload replied:
Yes but if it fails the first time, then doesn't on the second, if we don't set it to false then it will be classed as `failed` when it hasn't

ArsenArsen replied:
Oh correct, I forgot. Sorry. Exponential backoff

ArsenArsen commented on src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java:
```diff
@@ -72,4 +82,25 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
+            }
+        }
+        if (failed) {
+            throw new YoutubeAccessException();
```
Just use another built in exception to be honest, it's really the youtube state being bad, so IllegalStateException would work

Also in the future there's a big chance it won't be only youtube

binaryoverload replied:
As in we are going to hang other providers?

ArsenArsen replied:
Possible

ArsenArsen commented on src/main/resources/logback.xml:
```diff
@@ -7,8 +7,8 @@
     <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
-        <filter class="stream.flarebot.flarebot.util.ErrorCatcher"/>
         <filter class="stream.flarebot.flarebot.util.Filter"/>
+        <filter class="stream.flarebot.flarebot.util.ErrorCatcher"/>
```
?????

binaryoverload replied:
Needs to switch the order so it works properly

ArsenArsen replied:
Ah

ArsenArsen commented on src/main/java/stream/flarebot/flarebot/util/Filter.java:
```diff
@@ -19,6 +22,12 @@ public FilterReply decide(ILoggingEvent event) {
+        String thread = event.getThreadName();
+        for (String prefix : threadBlacklist) {
+            if (thread.startsWith(prefix)) {
+                return FilterReply.DENY;
```
Again, what's the point? The logs out of this thread are very useful often

binaryoverload replied:
You told me to get rid of them? In my last version I still logged them but stopped the errors. The reason I want the errors from the lava pool to not be passed on is because we can can catch them and even if we catch them the errors still log

ArsenArsen replied:
Yeah that was a mistake due to me being tired. And again they are more often that not useful.

binaryoverload replied:
Right, ok I'll revert back to what I did before

ArsenArsen commented:
Also why is this in data-restructure?

binaryoverload commented:
I explained the apache commons to walshy. the built in one in JDa doesn't work for us so I had to declare it in our POM.
This is in data-restructure to avoid merging conflicts from dev

ArsenArsen commented:
But how does it not work??????????????????????????????????

binaryoverload commented:
I don't know, I tested it and it just doesn't. It makes no difference really. It doesn't make our effective POM bigger or anything

ArsenArsen commented:
Elaborate

ArsenArsen commented on src/main/java/stream/flarebot/flarebot/music/extractors/YouTubeExtractor.java:
```diff
@@ -11,10 +11,14 @@
+import stream.flarebot.flarebot.util.GeneralUtils;
 import stream.flarebot.flarebot.util.MessageUtils;
 
+import java.awt.*;
```
Why all of AWT instead of just Color?

binaryoverload replied:
That's just IntelliJ importing * after more than 3 values are used

ArsenArsen replied:
I don't see the old ones removed which is why I asked

binaryoverload replied:
What do you mean by old ones?

ArsenArsen replied:
The old two imports that made it be 3?

binaryoverload replied:
Oh I see what you mean, I'll check that now

binaryoverload commented:
Basically, When I tried to run the test bot, it wouldn't run due to it's own commons-lang being overridden by our outdated one. SO I removed the outdated on and that side worked but then I couldn't use commons-lang in out project. So I added lang to our pom and both JDA and us used it and it worked.

ArsenArsen commented on src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java:
```diff
@@ -81,13 +81,18 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
+                try {
+                    Thread.sleep(backoff);
+                } catch (InterruptedException ignored) {}
+                backoff *= 2;
```
Supposed to be squared, so `backoff *= backoff`

binaryoverload replied:
Okie
